# Text-based Emotion Classifier

---

### Source of Dataset

> We will be using the [Emotions dataset for NLP](https://www.kaggle.com/praveengovi/emotions-dataset-for-nlp) by Praveen.

### Format of Dataset

>| text         | emotion |
>|--------------|---------|
>|i didnt feel humiliated | sadness |
>|i can go from feeling so hopeless to so damned hopeful just from being around... | sadness |
>|im grabbing a minute to post i feel greedy wrong | anger |
>|i am ever feeling nostalgic about the fireplace i will know that it is still... | love |

> **Note:** ***text*** and ***emotion*** are separated by a semi-colon ***';'***.
<br>
<pre>
i didnt feel humiliated;sadness
i am feeling grouchy;anger
...
</pre>

---

### Import Libraries & Dataset

We will start by importing the essential libraries.

> [Pandas](https://pandas.pydata.org/docs/) <br>
> [Numpy](https://numpy.org/doc/stable/) <br>
> [Seaborn](https://seaborn.pydata.org/tutorial.html) <br>
> [Matplotlib](https://matplotlib.org/3.5.1/)

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sb
import matplotlib.pyplot as plt

We will also import our train dataset `train.txt`.

In [ ]:
train_data = pd.read_csv("datasets/train.txt", sep = ';')
train_data.head()

text  emotion
0                            i didnt feel humiliated  sadness
1  i can go from feeling so hopeless to so damned...  sadness
2   im grabbing a minute to post i feel greedy wrong    anger
3  i am ever feeling nostalgic about the fireplac...     love
4                               i am feeling grouchy    anger

In [ ]:
train_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16000 entries, 0 to 15999
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   text     16000 non-null  object
 1   emotion  16000 non-null  object
dtypes: object(2)
memory usage: 250.1+ KB


In [ ]:
train_data.describe()

text emotion
count                                               16000   16000
unique                                              15969       6
top     i feel on the verge of tears from weariness i ...     joy
freq                                                    2    5362

Here are some observations from a quick glance through the dataset that was imported:
* There are a total of `16,000` data in the train dataset.
* There are a total of `6` different types of emotions in our train dataset.
* The `most common emotion` from the train dataset is `joy`.
* The emotion `joy` appeared `5362` times in our dataset.
* It appears that there exist multiple identical data in the train dataset as indicated by `count` > `unique`.
* We can also conclude that the identical data come in pairs as the top freq is `2`.

---